package/samplers/multi_armed_bandit/multi_armed_bandit.py

```diff
@@ -0,0 +1,63 @@
+from collections import defaultdict
+from dataclasses import dataclass
+from functools import partial
+from typing import Any
+from typing import Optional
+
+from optuna.distributions import BaseDistribution
+from optuna.samplers import RandomSampler
+from optuna.study import Study
+from optuna.study._study_direction import StudyDirection
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+@dataclass
+class ArmInfo:
+    value_avg: float
+    selected_num: int
+
+
+class MultiArmedBanditSampler(RandomSampler):
+    """Sampler based on Multi-armed Bandit Algorithm.
+
+    Args:
+        epsilon (float):
+            Params for epsolon-greedy algorithm.
+            epsilon is probability of selecting arm randomly.
+        seed (int):
+            Seed for random number generator and arm selection.
+
+    """
+
+    def __init__(
+        self,
+        epsilon: float = 0.7,
+        seed: Optional[int] = None,
+    ) -> None:
+        super().__init__(seed)
+        self._epsilon = epsilon
+        self._values: defaultdict = defaultdict(
+            lambda: defaultdict(partial(ArmInfo, 0, 0))
+        )  # param_name -> choice
+
+    def sample_independent(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        param_name: str,
+        param_distribution: BaseDistribution,
+    ) -> Any:
+        states = (TrialState.COMPLETE, TrialState.PRUNED)
+        trials = study._get_trials(deepcopy=False, states=states, use_cache=True)
+
+        rewards_by_choice: defaultdict = defaultdict(float)
```

Comment:
[QUESTION] This `defaultdict` treats never choiced arm having 0 rewards. Should i replace any other idea?
(as far as I can think of using `_n_startup_trials` like [TPESampler](https://github.com/ryota717/optuna/blob/326ac8d6402bafdb3d928a99e6ce799aeaed559a/optuna/samplers/_tpe/sampler.py#L407-L410),...) or letting user set initial reward)